controller/src/beerocks/master/tasks/task.cpp

```diff
@@ -89,6 +89,10 @@ void task::wait_for_task_end(int task_id, int ms)
     waiting                  = true;
     pending_task_id          = task_id;
 }
+void task::clear_pending_events()
```

Comment:
Please explain in the commit message why this is (or will be) needed so when we review this commit we understand the context. How come this was never needed so far?

Reply:
we clear events before task is restarted (ABORT or SCAN_FINISH) since if we dont handle_event() will not handle them and throw error in the log file.

maybe we should start using in other tasks also.

Reply:
will fix the commit comment after approval  in order not to break the order.

Reply:
done